app/models/account/states.rb

```diff
@@ -98,6 +100,20 @@ module Account::States
       scheduled_for_deletion.where.has { deleted_at <= value.ago }
     end
 
+    scope :without_traffic_since_time_ago, ->(value = TIME_TO_BE_INACTIVE) do
+      where.has do
+        not_exists Cinstance.where.has { user_account_id == BabySqueel[:accounts].id }.active_since_time_ago(value)
+      end
+    end
+
+    scope :inactive_since_time_ago, ->(value = TIME_TO_BE_INACTIVE) do
+      where.has { created_at <= value.ago }.without_traffic_since_time_ago(value)
+    end
+
+    scope :suspended_since_time_ago, ->(value = TIME_TO_SCHEDULE_FOR_DELETION) do
+      suspended.where.has { updated_at <= value.ago }
```

Comment:
Well, if you edit it (as a master, because the same account can't), it is because you don't want it removed, right? why would you edit something that you don't care about being removed? 😝

Reply:
Btw, if it is not completely locked (which i don't know), imo we should lock it (except changing the state) just like we do for the schedule for deletion, because otherwise it doesn't make much sense to have it suspended but working normally... 🤔

Reply:
It's the semantics that concerns me. We are assuming things here rather than making them explicit. This is easy to break IMO.

Reply:
@guicassolato Do you suggest is it better to have a `suspended_at` new field?

Reply:
I think we need to have some columns like `suspended_at` column like the `schedule_for_deletion_at` wdyt? @guicassolato @Martouta @macejmic 
I do not trust `updated_at` as it can be updated by any process in the background

Reply:
maybe create a `changed_state_at` and remove the `schedule_for_deletion_at` so we don't have 2 different fields for the last change of the state 🤔

Reply:
![image](https://user-images.githubusercontent.com/11318903/47740580-68f6ef80-dc78-11e8-8f4b-084f7536915a.png)

Reply:
Not a bad idea. We need to make sure that setting the changed_state_at happens at the same time as changing the state

Reply:
https://github.com/3scale/porta/issues/290 💪